src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hot_node/ThreadMetricsRca.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.rca.store.rca.hot_node;
+
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.opensearch.performanceanalyzer.PerformanceAnalyzerApp;
+import org.opensearch.performanceanalyzer.grpc.FlowUnitMessage;
+import org.opensearch.performanceanalyzer.metrics.AllMetrics;
+import org.opensearch.performanceanalyzer.metricsdb.MetricsDB;
+import org.opensearch.performanceanalyzer.rca.framework.api.Metric;
+import org.opensearch.performanceanalyzer.rca.framework.api.Rca;
+import org.opensearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import org.opensearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import org.opensearch.performanceanalyzer.rca.framework.api.metrics.Thread_Blocked_Time;
+import org.opensearch.performanceanalyzer.rca.framework.api.metrics.Thread_Waited_Time;
+import org.opensearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import org.opensearch.performanceanalyzer.rca.framework.core.RcaConf;
+import org.opensearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import org.opensearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+
+/**
+ * Analyzes thread_blocked_time and thread_waited_time metrics for transport threads and publishes
+ * the following data: 1. Number of threads which are blocked for more than 10 seconds in past
+ * minute. 2. Number of threads which are in waiting/timed-waiting state for more than 10 seconds in
+ * past minute. 3. Maximum value of thread_blocked_time for a single thread in the past minute. 4.
+ * Maximum value of thread_waited_time for a single thread in the past minute.
+ */
+public class ThreadMetricsRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
```

Comment:
Why can't we generalize this to other threads as well instead of just transport threads? We can just pass the list of thread names we need to analyze during initialization and then accordingly calculate. This makes it more extensible for future use cases if needed.